backend/hitas/calculations/max_prices/rules_pre_2011.py

```diff
@@ -42,6 +44,11 @@ def calculate_construction_price_index_max_price(
         housing_company_improvements: List,
         calculation_date: datetime.date,
     ) -> IndexCalculation:
+        if not apartment.completion_date_realized_housing_company_acquisition_price:
```

Comment:
Should also check `realized_housing_company_acquisition_price` is not zero/null since `calculate_housing_company_improvements_pre_2011_construction_price_index` cannot be performed without it (leads to 0 division error).